### ISSUE TITLE
Netty is not shutting down cleanly after EtcdClient.close is called

### ISSUE THREAD

**Issue by oakad** [jurmous/etcd4j#61]
I have a simple etcd4j based application which uses the wait facility to wait on Etcd key changes. When shutting down the application (EtcdClient.close() is called from within proto.close()) the Netty won't shutdown cleanly, even if I take care to explicitly cancel all promises beforehand.

It appears that EtcdClient is not trying hard enough to shutdown the underlying Netty factories or whatever.

proto.close()
[main] INFO mousio.etcd4j.transport.EtcdNettyClient - Shutting down Etcd4j Netty client
after proto.close()
[pool-1-thread-1] WARN io.netty.util.HashedWheelTimer - An exception was thrown by TimerTask.
java.lang.IllegalStateException: executor not accepting a task
	at io.netty.resolver.NameResolverGroup.getResolver(NameResolverGroup.java:58)
	at io.netty.bootstrap.Bootstrap.doResolveAndConnect(Bootstrap.java:164)
	at io.netty.bootstrap.Bootstrap.connect(Bootstrap.java:139)
	at io.netty.bootstrap.Bootstrap.connect(Bootstrap.java:120)
.... 10 pages of the usual Netty dung omitted

**Comment by valderrama** [comment]
Seeing the same issue where the client closes but the process doesn't end because there's still a non-daemon thread hanging around, I believe [line 18 in RetryPolicy](https://github.com/jurmous/etcd4j/blob/c718b0f4405ae448fe89e7928c134555557470fc/src/main/java/mousio/client/retry/RetryPolicy.java#L18) is the culprit:

```java
private static final HashedWheelTimer timer = new HashedWheelTimer();
```

Per the [Netty documentation](http://netty.io/4.0/api/io/netty/util/HashedWheelTimer.html#stop()) even though start was never explicitly called, stop() should be called on the timer object to clean up all the resources including this thread pool (from my jstack trace):

```
"pool-1-thread-1" #12 prio=0 os_prio=0 tid=0x0000 nid=0x0000 waiting on condition [0x0000]
   java.lang.Thread.State: TIMED_WAITING (sleeping)
	at java.lang.Thread.sleep(Native Method)
	at io.netty.util.HashedWheelTimer$Worker.waitForNextTick(HashedWheelTimer.java:461)
	at io.netty.util.HashedWheelTimer$Worker.run(HashedWheelTimer.java:360)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by jurmous** [comment]
Thanks for the reporting in! I did not close HashedWheelTimer because it can be used by multiple etcd clients. It is advised to only have one in a App. 
I now catch retried calls so they are canceled if they are still open if event loop is closed.

**Comment by jurmous** [comment]
It is fixed in the 2.8 release.
